test/metabase/api/action_test.clj

```diff
@@ -206,3 +211,71 @@
               (is (partial= {:errors {:error_handle "nullable string, and must be a valid json-query, something like '.item.title'"},
                              :specific-errors {:error_handle ["must be a valid json-query, something like '.item.title'"]}}
                             (mt/user-http-request :crowberto :put 400 action-path (assoc initial-action :error_handle "x")))))))))))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                            PUBLIC SHARING ENDPOINTS                                            |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(def ^:private unshared-action-opts
+  {:public_uuid       nil
+   :made_public_by_id nil})
+
+(defn- shared-action-opts []
+  {:public_uuid       (str (UUID/randomUUID))
+   :made_public_by_id (mt/user->id :crowberto)})
+
+(deftest share-action-test
+  (testing "POST /api/action/:id/public_link"
+    (mt/with-temporary-setting-values [enable-public-sharing true]
+      (actions.test-util/with-actions-enabled
+        (mt/with-non-admin-groups-no-root-collection-perms
+          (testing "We can share an action"
+            (actions.test-util/with-actions [{:keys [action-id]} unshared-action-opts]
+              (let [uuid (:uuid (mt/user-http-request :crowberto :post 200
+                                                      (format "action/%d/public_link" action-id)))]
+                (is (db/exists? Action :id action-id, :public_uuid uuid))
+                (testing "Test that if an Action has already been shared we reüse the existing UUID"
```

Comment:
```suggestion
                (testing "Test that if an Action has already been shared we reuse the existing UUID"
```
?

(I thought this might be a *The New Yorker*-style orthography decision, but they only do it for words with repeated but re-enunciated vowels like coöperate (ahem, reënunciated))

Reply:
Ha! Nice spotting